### PR TITLE
support for NETCONF messages

### PR DESCRIPTION
**Pull request by michalvasko**

### CODE REVIEW

File: src/parser_data.h [hunk 1]
```diff
@@ -187,53 +187,33 @@ struct ly_in;
...
 /**
  * @brief Parse (and validate) data from the input handler as a YANG data tree.
  *
  * @param[in] ctx Context to connect with the tree being built here.
+ * @param[in] parent Optional parent to connect the parsed nodes to.
  * @param[in] in The input handle to provide the dumped data in the specified @p format to parse (and validate).
  * @param[in] format Format of the input data to be parsed. Can be 0 to try to detect format from the input handler.
  * @param[in] parse_options Options for parser, see @ref dataparseroptions.
  * @param[in] validate_options Options for the validation phase, see @ref datavalidationoptions.
- * @param[out] tree Resulting data tree built from the input data. Note that NULL can be a valid result as a representation of an empty YANG data tree.
```
Comment (rkrejci) [inline src/parser_data.h:217]: Why aren't these notes valid now?

**Review by rkrejci — CHANGES_REQUESTED** [review]
Please review also the transition manual text, there are probably parts to update/add.

File: src/in.c [hunk 2]
```diff
@@ -380,51 +380,107 @@ lyd_ctx_free(struct lyd_ctx *lydctx)
     ly_set_erase(&lydctx->node_when, NULL);
 }
 
+LY_ERR
+lyd_parser_find_operation(const struct lyd_node *parent, uint32_t int_opts, struct lyd_node **op)
+{
+    const struct lyd_node *iter;
+
```
[the remark this answers is not in the document]
Comment (michalvasko) [reply src/in.c:387]: It should already be `NULL` but sure, why not.

File: src/parser_data.h [hunk 3]
```diff
@@ -243,85 +223,99 @@ LY_ERR lyd_parse_data_mem(const struct ly_ctx *ctx, const char *data, LYD_FORMAT
...
- * @param[in] ctx Context to connect with the tree being parsed.
- * @param[in] in Input handle to provide the dumped data in the specified @p format to parse.
- * @param[in] format Format of the input data to be parsed.
- * @param[out] tree Resulting full RPC/action request tree built from the input data. The returned data are expected to be freed using ::lyd_free_all().
- * In contrast to YANG data tree, result of parsing RPC/action cannot be NULL until an error occurs.
- * @param[out] op Optional pointer to the actual operation node inside the full action @p tree, useful only for action.
- * @return LY_SUCCESS in case of successful parsing.
- * @return LY_ERR value in case of error. Additional error information can be obtained from the context using ly_err* functions.
- */
-LY_ERR lyd_parse_rpc(const struct ly_ctx *ctx, struct ly_in *in, LYD_FORMAT format, struct lyd_node **tree,
```
[the remark this answers is not in the document]
Comment (michalvasko) [reply src/parser_data.h:291]: Right, forgot about that, fixed.

File: src/parser_data.h (hunk 1 above)
Comment (michalvasko) [reply src/parser_data.h:217]: I thought there was some useless information so I shortened it, maybe too much, I have added this note back.

**Review by rkrejci — APPROVED** [review]

**Pull request merged**